### PR TITLE
URLHaus - adding retry mechanism

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/24277
fixes: https://github.com/demisto/etc/issues/23857

## Description
URLHaus - adding retry mechanism to HTTP request and a param for it.

### CODE REVIEW

**Comment by roysagi** [comment]
@kirbles19 - If you could please have a look at the:
* [CHANGELOG](https://github.com/demisto/content/blob/urlhaus-fix/Packs/URLHaus/Integrations/URLHaus/CHANGELOG.md)
* [yml file](https://github.com/demisto/content/blob/urlhaus-fix/Packs/URLHaus/Integrations/URLHaus/URLHaus.yml) - specifically lines 28-30

**Comment by roysagi** [comment]
@David-BMS - please confirm that it ok to merge - DBotScore wise

**Comment by roysagi** [comment]
@ronykoz - design team has decided that no DBotScore will be returned for files - so this will fail on validation and require force merge.

**Comment by ronykoz** [comment]
@roysagi ping me once the build finishes

**Comment by Itay4** [comment]
does that fix also https://github.com/demisto/etc/issues/23857 ?

**Comment by roysagi** [comment]
> does that fix also [demisto/etc#23857](https://github.com/demisto/etc/issues/23857) ?

It might. I will take it out of skip to test out

**Comment by roysagi** [comment]
@ronykoz - Circle fails on the lack of DBotScore as expected. 

Waiting for @kirbles19 and then this needs a force merge.

File: Packs/URLHaus/Integrations/URLHaus/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ## [Unreleased]
-
+- Added *Number of retries* instance parameter to determine how many times should a command be attempted before raising an error.
```
Comment (amshamah419) [inline Packs/URLHaus/Integrations/URLHaus/CHANGELOG.md:2]:
```suggestion
- Added the *Number of retries* parameter which determines how many times a command should attempt before raising an error.
```

File: Packs/URLHaus/Integrations/URLHaus/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
...
+- Fixed an issue where ***urlhaus-download-sample*** command would raise an error in case on results were found.
```
Comment (amshamah419) [inline Packs/URLHaus/Integrations/URLHaus/CHANGELOG.md:3]:
```suggestion
- Fixed an issue where the ***urlhaus-download-sample*** command would raise an error in cases where results were found.
```

**Pull request merged**